chia-blockchain/templates/systemd/chia-data-layer.service.j2

```diff
@@ -8,7 +8,7 @@ PartOf={{ chia_service_base }}.target
 Type=simple
 # Check to see if wallet is responding before starting. Data layer will fail
 # if the wallet service does not respond.  Hard-coded to localhost.
-ExecStartPre=/bin/bash -c '(while ! nc -z -v -w1 localhost 9256 2>/dev/null; do echo "Waiting for the wallet to listen on port 9256..."; sleep 1; done); sleep 1'
+ExecStartPre=/bin/bash -c '(while ! nc -z -v -w1 localhost {{ chia_wallet_rpc_port }} 2>/dev/null; do echo "Waiting for the wallet to listen on port {{ chia_wallet_rpc_port }}..."; sleep 1; done); sleep 1'
```

Comment:
@TheLastCicada I wonder if we should move this to the wallet service as an `ExecStartPost`, so that the wallet service just isn't marked as ready until the port is up? Advantage would be that any other service that requires it automatically inherits this check, vs needing to copy paste everywhere

Reply:
Moved this to the wallet. Let me know what you think